core/services/pipeline/task.eth_tx.go

```diff
@@ -229,11 +229,11 @@ func setJobIDOnMeta(lggr logger.Logger, vars Vars, meta *txmgr.EthTxMeta) {
 	if err != nil {
 		return
 	}
-	jobIDF, is := jobID.(float64) // JSON decoder default numeric type
-	if is {
-		jobIDInt := int32(jobIDF)
-		meta.JobID = &jobIDInt
-	} else {
+	switch v := jobID.(type) {
+	case int64:
```

Comment:
We don't handle other cases here, because the target variable is of type int32 anyway.